src/lzfse_decode.c

```diff
@@ -48,3 +50,24 @@ size_t lzfse_decode_buffer(uint8_t *__restrict dst_buffer, size_t dst_size,
     return 0;                           // failed
   return (size_t)(s->dst - dst_buffer); // bytes written
 }
+
+size_t lzfse_decode_buffer(uint8_t *__restrict dst_buffer, size_t dst_size,
+                           const uint8_t *__restrict src_buffer,
+                           size_t src_size, void *__restrict scratch_buffer) {
+  int has_malloc = 0;
+  size_t ret = 0;
+
+  // Deal with the possible NULL pointer
+  if (scratch_buffer == NULL) {
+    scratch_buffer = malloc(lzfse_encode_scratch_size());
```

Comment:
it's assuming scratch size is non 0. It's OK now, but may introduce a latent bug if it changes in the future.